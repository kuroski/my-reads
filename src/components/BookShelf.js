import React from 'react';
import PropTypes from 'prop-types';
import Book from './Book';

const BookShelf = props => {
  const renderBooks = () => {
    if (!props.books || props.books.length === 0)
      return <li className="empty-message">No books on this shelf</li>;

    return props.books.map(book => (
      <li key={book.id}>
        <Book
          id={book.id}
          title={book.title}
          authors={book.authors}
          coverImage={book.coverImage}
          onMove={props.onMove}
          selectedShelf={props.shelfId}
          shelves={props.shelves}
        />
      </li>
    ));
  };

  return (
    <div className="bookshelf">
      <h2 className="bookshelf-title">{props.name}</h2>
      <div className="bookshelf-books">
        <ol className="books-grid">{renderBooks()}</ol>
      </div>
    </div>
  );
};

BookShelf.propTypes = {
  name: PropTypes.string.isRequired,
  shelfId: PropTypes.string,
  onMove: PropTypes.func.isRequired,
  books: PropTypes.array,
  shelves: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string,
      value: PropTypes.string
    })
  )
};

BookShelf.defaultProps = {
  books: [],
  shelves: [],
  shelfId: ''
};

export default BookShelf;
