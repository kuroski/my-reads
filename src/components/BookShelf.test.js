import React from 'react';
import renderer from 'react-test-renderer';
import BookShelf from './BookShelf';

describe('BookShelf Container', () => {
  it('renders correctly', () => {
    const BookShelfTree = renderer.create(<BookShelf />).toJSON();
    expect(BookShelfTree).toMatchSnapshot();
  });
});
