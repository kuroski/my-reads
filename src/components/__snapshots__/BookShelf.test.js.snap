// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookShelf Container renders correctly 1`] = `
<div
  className="bookshelf"
>
  <h2
    className="bookshelf-title"
  >
    Currently Reading
  </h2>
  <div
    className="bookshelf-books"
  >
    <ol
      className="books-grid"
    >
      <li>
        <div
          className="book"
        >
          <div
            className="book-top"
          >
            <img
              alt="1776"
              className="book-cover"
              height="193"
              src="http://books.google.com/books/content?id=uu1mC6zWNTwC&printsec=frontcover&img=1&zoom=1&imgtk=AFLRE73pGHfBNSsJG9Y8kRBpmLUft9O4BfItHioHolWNKOdLavw-SLcXADy3CPAfJ0_qMb18RmCa7Ds1cTdpM3dxAGJs8zfCfm8c6ggBIjzKT7XR5FIB53HHOhnsT7a0Cc-PpneWq9zX&source=gbs_api"
              width="128"
            />
            <div
              className="book-shelf-changer"
            >
              <select
                onChange={[Function]}
              >
                <option
                  disabled={true}
                  value="none"
                >
                  Move to...
                </option>
                <option
                  value="currentlyReading"
                >
                  Currently Reading
                </option>
                <option
                  value="wantToRead"
                >
                  Want to Read
                </option>
                <option
                  value="read"
                >
                  Read
                </option>
                <option
                  value="none"
                >
                  None
                </option>
              </select>
            </div>
          </div>
          <div
            className="book-title"
          >
            1776
          </div>
          <div
            className="book-authors"
          >
            Test author, Daniel
          </div>
        </div>
      </li>
    </ol>
  </div>
</div>
`;
